Assets/Scripts/PlayerCharacter.cs

```diff
@@ -0,0 +1,256 @@
+        }
+        anim.SetBool("Falling", falling);
+        anim.SetFloat("Speed", Math.Abs(myRigidBody.velocity.x)); //lets the animator know the speed of player.
+    }
+
+    private void FixedUpdate()
+    {
+        UpdatePhysicsMaterial();
+        if (Time.time - deathTimer >= timeToDeath + 0.5f || deathTimer == 0)
+        {
+            //character cannot use the move function when dying.
+            if (Input.GetKey(KeyCode.D) || Input.GetKey(KeyCode.A))
+            {
+                Move();//lets player move a fixed amount not dependent on frames
+            }
+            //player flips sprite when going opposite direction of prior movement.
+            if (direction > 0 && !facingRight)
+            {
+                Flip();
+            }
+            else if (direction < 0 && facingRight)
+            {
+                Flip();
+            }
+            deathTimer = 0;
+        }
+    }
+
+    private void UpdatePhysicsMaterial()
+    {
+        //changes physics material when moving or stopping.
+        if (Mathf.Abs(direction) > 0)
+        {
+            playerGroundCollider.sharedMaterial = playerMovingPhysicsMaterial;
+        }
+        else
+        {
+            playerGroundCollider.sharedMaterial = playerStoppingPhysicsMaterial;
+        }
+    }
+
+    private void GetMovementInput()
+    {
+        //sets direction of movement from player commands
+        direction = Input.GetAxisRaw("Horizontal");
+    }
+
+    private void Move()
+    {
+        myRigidBody.AddForce(Vector2.right * direction * acceleration / speedJumpReducer);
+        Vector2 clampedVelocity = myRigidBody.velocity;
+        clampedVelocity.x = Mathf.Clamp(myRigidBody.velocity.x, -maxSpeed / speedJumpReducer, maxSpeed / speedJumpReducer);
+        /*if (isInDeath)
+        {
+            clampedVelocity.x = clampedVelocity.x/10;
+        }*/
+        myRigidBody.velocity = clampedVelocity;//adds up a build up of acceleration to a maximum velocity
+    }
+
+    private void Jump()
+    {
+        //Allows player to jump twice before having to fall to the ground.
+       if (jumpNumber == 0&&grounded==true)
+        {
+            audioSource.Play();
+            anim.SetBool("IsOnGround", false);
+            myRigidBody.velocity = new Vector2(myRigidBody.velocity.x, jumpOneModifier);
+            jumpNumber++;
+            jumpCooldown = 50;
+        }
+        else if (jumpNumber == 1) {
+            audioSource.Play();
+            myRigidBody.velocity = new Vector2(myRigidBody.velocity.x, jumpTwoModifier);
+            jumpCooldown = 90;
+            jumpNumber++;
+        }
+        
+    }
+
+    public void StartRespawn()
+    {
+        //stage one of respawn
+        isInDeath = true;
+        deathTimer = Time.realtimeSinceStartup;
+    }
+
+    private void Respawn()
+    {
+        //last stage of respawn, checks for checkpoint and resets 'is dead' to default of false
+        anim.SetBool("IsDead", false);
```

Comment:
move to your `UpateAnimationParameters` function.

Reply:
Don't hardcode it, have an `isDead` bool that is up to date and set the anim param to `isDead` in your `UpdateAnimParameters`